go/libraries/doltcore/schema/encoding/schema_marshaling.go

```diff
@@ -316,6 +316,9 @@ func (sd schemaData) addChecksIndexesAndPkOrderingToSchema(sch schema.Schema) er
 func MarshalSchemaAsNomsValue(ctx context.Context, vrw types.ValueReadWriter, sch schema.Schema) (types.Value, error) {
 	// Anyone calling this is going to serialize this to disk, so it's our last line of defense against defective schemas.
 	// Business logic should catch errors before this point, but this is a failsafe.
+
+	schema.RemoveDuplicateNotNullColumnConstraints(sch.GetAllCols())
```

Comment:
Let's validate the schema as we write it, and clean the schema as we read it. Two reasons
- Validate-on-write forces us to make the correct operations as we're modifying the schema, rather than catching errors here at the last step. 
- Clean-on-read allows us to correct the mistakes that currently exists in any Dolt databases "in the wild".